### ISSUE TITLE
[Bug]: Bug while creating a PodMonitor

### ISSUE THREAD

**Issue by CiraciNicolo** [cloudnative-pg/cloudnative-pg#4286]
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

nicolo.ciraci@elmec.it

### Version

trunk (main)

### What version of Kubernetes are you using?

1.29

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

Other

### What happened?

While reviewing the PodMonitor functionality as described in #4268, we found a regression caused by the new Plugin API implementation that crashed the controller while creating the PodMonitor. The issue can be found [here](https://github.com/cloudnative-pg/cloudnative-pg/blob/6e423eb50b71ced1fe78aaecdb1cf0411aefdac5/internal/cnpi/plugin/client/lifecycle.go#L61).

The new lifecycle hook does not recognise the new PodMonitor CR and return nil to the Kubernetes API client; this triggers a reconcile error in which the cluster is not created. 

A fix is implemented in the PR #4282.

### Cluster resource

```shell
---
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: cluster-default-monitoring
spec:
  instances: 1

  monitoring:
    enablePodMonitor: true

  storage:
    size: 1Gi
```

### Relevant log output

```shell
{"level":"error","ts":"2024-04-15T14:17:53Z","msg":"Reconciler error","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"cluster-default-monitoring","namespace":"default"},"namespace":"default","name":"cluster-default-monitoring","reconcileID":"47f777a3-d2f2-4ed0-8651-0444a5a422ec","error":"cannot create Cluster auxiliary objects: expected pointer, but got invalid kind","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/nicolo.ciraci/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/nicolo.ciraci/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/Users/nicolo.ciraci/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:227"}
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by rouke-broersma** [comment]
Any workaround so I can enable pod monitors on my existing databases?

**Comment by CiraciNicolo** [comment]
We suggest to disable the built-in podMonitor functionality, and define your own podMonitor.

**Comment by rouke-broersma** [comment]
@CiraciNicolo yes it is, I work around it by deploying my own podmonitor.

**Comment by Ornias1993** [comment]
> We suggest to disable the built-in podMonitor functionality, and define your own podMonitor.
> 
> @rouke-broersma is the regression present in the latest release?

More importantly:
Why isn't this covered by your testing suite?

**Comment by CiraciNicolo** [comment]
> > We suggest to disable the built-in podMonitor functionality, and define your own podMonitor.
> 
> > 
> 
> > @rouke-broersma is the regression present in the latest release?
> 
> 
> 
> More importantly:
> 
> Why isn't this covered by your testing suite?

I'm not involved in the cnpg project, so I didn't know 🤗 

I was talking on behalf of me and my colleagues, which are using this project for some internal database. 

Furthermore when we discovered this issue, looking into other improvements we proposed to the maintainers we committed test dedicated to this issue - as you can see in the diffs.

**Comment by Ornias1993** [comment]
> I was talking on behalf of me and my colleagues, which are using this project for some internal database.

Okey, best not to talk in "we" unless people know who "we" is.

> Furthermore when we discovered this issue, looking into other improvements we proposed to the maintainers, we committed tests dedicated to this issue - as you can see in the diffs.

What diffs what PR's?

**Comment by CiraciNicolo** [comment]
> > Furthermore when we discovered this issue, looking into other improvements we proposed to the maintainers, we committed tests dedicated to this issue - as you can see in the diffs.
> 
> 
> 
> What diffs what PR's?

The PR is linked in this bug's description:

> A fix is implemented in the PR #4282.

**Comment by Ornias1993** [comment]
> > > Furthermore when we discovered this issue, looking into other improvements we proposed to the maintainers, we committed tests dedicated to this issue - as you can see in the diffs.
> > 
> > 
> > What diffs what PR's?
> 
> The PR is linked in this bug's description:
> 
> > A fix is implemented in the PR #4282.

Shoot i'm blind sorry!
Thanks for the PR
